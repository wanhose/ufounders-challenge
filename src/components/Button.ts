import styled from 'styled-components';

const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.ufounders100};
  border: none;
  border-radius: 0.5rem;
  color: ${({ theme }) => theme.colors.trenchWhite};
  cursor: pointer;
  font-family: ${({ theme }) => theme.fonts.poppins};
  font-size: 1rem;
  font-weight: ${({ theme }) => theme.weights.semiBold};
  line-height: 1.5rem;
  padding: 0.75rem 1rem;
  transition: background-color linear 0.5s;

  :focus {
    background-color: ${({ theme }) => theme.colors.ufounders120};
  }

  :hover {
    background-color: ${({ theme }) => theme.colors.ufounders80};
  }

  :disabled {
    background-color: ${({ theme }) => theme.colors.ufounders20};
    cursor: not-allowed;
  }
`;

export default Button;
