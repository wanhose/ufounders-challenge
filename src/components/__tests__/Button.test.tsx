import { render } from 'tests/utils';

import Button from '../Button';

describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(<Button />);

    expect(container).toMatchSnapshot('Regular');
  });
});
