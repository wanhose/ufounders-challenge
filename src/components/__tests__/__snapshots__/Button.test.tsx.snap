// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly: Regular 1`] = `
.c0 {
  background-color: #e93256;
  border: none;
  border-radius: 0.5rem;
  color: #ffffff;
  cursor: pointer;
  font-family: Poppins;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5rem;
  padding: 0.75rem 1rem;
  -webkit-transition: background-color linear 0.5s;
  transition: background-color linear 0.5s;
}

.c0:focus {
  background-color: #d6284a;
}

.c0:hover {
  background-color: #e55373;
}

.c0:disabled {
  background-color: #f8d4dc;
  cursor: not-allowed;
}

<div>
  <button
    class="c0"
  />
</div>
`;
